src/main/scala/util/HeterogeneousBag.scala

```diff
@@ -6,12 +6,34 @@ import Chisel._
 import chisel3.Record
 import scala.collection.immutable.ListMap
 
-final case class HeterogeneousBag[T <: Data](elts: Seq[T]) extends Record with collection.IndexedSeq[T] {
+final case class HeterogeneousBag[T <: Data](elts: Seq[T], eltNames: Option[Seq[String]] = None)
+    extends Record with collection.IndexedSeq[T] {
```

Comment:
I know it was already like this, but mixing in `IndexedSeq` here is not true because "Indexed sequences support constant-time or near constant-time element access and length computation." (https://www.scala-lang.org/api/2.12.10/scala/collection/IndexedSeq.html). It's not a huge deal, just misleading about performance, I think it would be better to change this to extending `Seq`.

Reply:
The observation that this is not upholding the API is that `apply: Int => T` and `length` below both delegate to `elts` which is itself just a `Seq`.

Reply:
ok, this may become irrelevant if i end up making a new class, do you still think i should fix this here?